app/login/login.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { Http, Headers, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+
+@Injectable()
+export class LoginService {
+  constructor(private http: Http) { }
```

Comment:
private variables are visible within the class only and during the compiling these are hide while using other classes and thus performance improves